src/file-uploader/file-uploader.stories.ts

```diff
@@ -84,6 +86,122 @@ class FileUploaderStory {
 	}
 }
 
+@Component({
+	selector: "app-drop-file-uploader",
+	template: `
+		<ibm-file-uploader
+			[title]="title"
+			[description]="description"
+			[buttonText]="buttonText"
+			[buttonType]="buttonType"
+			[accept]="accept"
+			[multiple]="multiple"
+			[skeleton]="skeleton"
+			[(files)]="files"
+			[size]="size"
+			drop="true"
+			[dropText]="dropText"
+			(filesChange)="onDropped($event)">
+		</ibm-file-uploader>
+
+		<div [id]="notificationId" style="width: 300px; margin-top: 20px"></div>
+		<button ibmButton *ngIf="files && files.size > 0" (click)="onUpload()">
+			Upload
+		</button>
+	`
+})
+class DragAndDropStory {
+	static notificationCount = 0;
+
+	@Input() notificationId = `notification-${FileUploaderStory.notificationCount}`;
+	@Input() files = new Set();
+	@Input() title;
+	@Input() description;
+	@Input() accept = [".jpg", ".png"];
+	@Input() multiple;
+	@Input() dropText = "Drag and drop files here of upload";
+
+	protected maxSize = 500000;
+
+	constructor(protected notificationService: NotificationService) {
+		FileUploaderStory.notificationCount++;
+	}
+
+	// This is an example of further filtration which can take place after
+	// preliminary filtration and is optional.
+	onDropped(event) {
+		const transferredFiles = Array.from(event);
+
+		const fileExtensionRegExp = new RegExp(/\.[0-9a-z]+$/, "i");
+
+		let promises: Array<Promise<any>> = [];
+
+		// Creates a promise which resolves to a file and whether or not the file should be accepted.
+		const readFileAndAddToMap = fileObj => {
+			return new Promise((resolve, reject) => {
+				let fileExtension, mime;
+				let reader = new FileReader();
+				reader.readAsArrayBuffer(fileObj.file);
+				reader.onload = () => {
+					// Checks the type of file based on magic numbers.
+					const type = fileType(reader.result);
+					if (type) {
+						fileExtension = type.ext.replace(/^/, ".");
+						mime = type.mime;
+					} else {
+						// If a file type can not be determined using magic numbers
+						// then use file extension or mime type.
+						[fileExtension] = fileObj.file.name.match(fileExtensionRegExp);
+						mime = fileObj.file.type;
+					}
```

Comment:
This logic isn't totally clear ...

Reply:
If a file type can not be determined using magic numbers the fileType method will return undefined. So in that case it uses the file extension or mime type. Would a comment be enough to explain that or should I rethink the logic?

Reply:
I just realized that the else block is the exact same as the preliminary filter in the component. In that case would it be better to just accept the file automatically if it reaches that point?

Reply:
Most likely, seems redundant otherwise